src/senaite/sync/fetchstep.py

```diff
@@ -85,29 +86,38 @@ def _fetch_data(self, window=1000, overlap=10):
         ordered_uids = storage["ordered_uids"]
         self.sh = SoupHandler(self.domain_name)
         # Dummy query to get overall number of items in the specified catalog
-        catalog_data = self.get_json("search", catalog='uid_catalog', limit=1)
+        query = {
+            "url_or_endpoint": "search",
+            "catalog": 'uid_catalog',
+            "limit": 1
+        }
+        if self.content_types:
+            query["portal_type"] = self.content_types
+        cd = self.get_json(**query)
         # Knowing the catalog length compute the number of pages we will need
         # with the desired window size and overlap
         effective_window = window-overlap
-        number_of_pages = (catalog_data["count"]/effective_window) + 1
+        number_of_pages = (cd["count"]/effective_window) + 1
```

Comment:
Why this? Wasn't there once a method `yield_items` that exactly went through the pages?

Reply:
Yes there was and there is, thanks to you : )
But since we retrieve the objects in the order of modification time ('uid_catalog' does it actually), considering that some objects can be modified in the source during `fetch` process in the destination, we query some records twice. I.e: First query retrieves records from '0-1000' and the second query '990-1990' and so on. So if there are 10 objects being modified during each fetch query, we are still safe.

Reply:
Ah, tricky :) But what if 11 objects were modified 🙈That was the reason I suggested that we just fetch the UID, portal_type and physical path in step 1 to just be able to create the object slugs. Then in step 2 I would fetch just in time to update the objects and probably re-fetch any missing objects...

Reply:
We already fetch just UID, portal_type and path in first step...You mean fetching all the objects at once? It can be done [here](https://github.com/senaite/senaite.sync/blob/master/src/senaite/sync/fetchstep.py#L73) by increasing window size. But then with a source with 800.000 objects, it was loading memory to the full, especially if the source and the destination is on the same machine.
Please let me know if I didn't clearly understand what you had meant.